gateway-ha/src/main/java/com/lyft/data/gateway/ha/clustermonitor/FaultTolerantHealthChecker.java

```diff
@@ -0,0 +1,60 @@
+package com.lyft.data.gateway.ha.clustermonitor;
+
+import com.lyft.data.gateway.ha.notifier.Notifier;
+
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+public class FaultTolerantHealthChecker extends HealthChecker {
+
+    public static final int DEFAULT_CONSECUTIVE_FAILURE_ALERT_COUNT = 5;
```

Comment:
IMO, alert-deduping logic should be abstracted in Notifier instead of HealthChecker, No?

Reply:
Yes, it is likely better for the health checker to only publish metrics, and define the alert rules in external systems.